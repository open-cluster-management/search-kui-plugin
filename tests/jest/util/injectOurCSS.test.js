/*******************************************************************************
* Licensed Materials - Property of IBM
* (c) Copyright IBM Corporation 2019. All Rights Reserved.
*
* Note to U.S. Government Users Restricted Rights:
* Use, duplication or disclosure restricted by GSA ADP Schedule
* Contract with IBM Corp.
*******************************************************************************/
'use strict'

const CSS = require('../../../dist/src-web/util/injectOurCSS')
const env = 'test'

describe('Injecting our CSS', () => {
  const spy = jest.spyOn(CSS, 'injectOurCSS')
  let _ = CSS.injectOurCSS(env)

  it(`should import and inject our css for environment (${env})`, () => {
    expect(CSS.injectOurCSS(env)).toBeDefined()
    expect(spy).toBeCalled()
    expect(CSS.injectOurCSS(env)).toMatchSnapshot()
  })

  _ = CSS.injectOurCSS('dev')

  it(`should import and inject our css for environment (dev)`, () => { // Will return undefined, due to issues accessing '@kui-shell/plugin-search/mdist/src-web/styles/index.css'
    expect(spy).toBeCalled()
    expect(CSS.injectOurCSS('dev')).toMatchSnapshot()
  })
})