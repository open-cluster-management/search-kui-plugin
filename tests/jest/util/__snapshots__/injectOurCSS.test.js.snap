// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Injecting our CSS should import and inject our css for environment (dev) 1`] = `undefined`;

exports[`Injecting our CSS should import and inject our css for environment (test) 1`] = `
Object {
  "css": "@kui-shell/plugin-search/mdist/src-web/styles/index.css",
  "key": "search",
}
`;
